testing/base.py

```diff
@@ -1,74 +1,73 @@
+import os
+import shutil
 import unittest
-from importlib import reload
-import subprocess
 
 from six import with_metaclass
 
-from .mocks import SphinxAppMock
-from .utils import cd, deep_update
+import pytest
+
+
+TEST_DOC_DIR = os.path.join(os.path.dirname(__file__), 'doc')
+
+
+def make_default_config(project):
+    """
+    Creates a default configuration for exhale
+    """
+    return {
+        'breathe_projects': {
+            project: './_doxy/xml'
+        },
+        'breathe_default_project': project,
+        'exhale_args': {
+            'containmentFolder': './api',
+            'rootFileName': 'index.rst',
+            'rootFileTitle': 'API Documentation',
+            'doxygenStripFromPath': '..',
+            'exhaleExecutesDoxygen': True,
+            'exhaleDoxygenStdin': 'INPUT = ../projects/%s/include' % project
+        }
+    }
 
 
 class ExhaleTestCaseMetaclass(type):
     """
     Metaclass to enforce mandatory attributes on ExhaleTestCase
     """
 
-    def __new__(cls, name, bases, attrs):
-        if attrs['__module__'] != 'testing.base' and attrs.get('test_project', None) is None:
+    def __new__(mcs, name, bases, attrs):
+        if attrs['__module__'] != __name__ and attrs.get('test_project', None) is None:
             raise RuntimeError('ExhaleTestCase subclasses must define a "test_project" attribute')
-        return super(ExhaleTestCaseMetaclass, cls).__new__(cls, name, bases, attrs)
+
+        has_tests = False
+
+        for name, attr in attrs.items():
+            if callable(attr) and name.startswith('test_'):
+                attrs[name] = pytest.mark.sphinx(
+                    testroot=TEST_DOC_DIR,
+                    confoverrides=attrs.get('config', None) or make_default_config(attrs['test_project'])
```

Comment:
actually we may still want to 'deep_update' the default config with attrs['config']. I may have removed the ``utils`` module a bit too quickly.

Reply:
Yes I was actually using it, in my local stashed changes it has been revived :)